lib/ClangImporter/ImportDecl.cpp

```diff
@@ -10004,19 +9997,34 @@ Decl *ClangImporter::Implementation::importDeclAndCacheImpl(
   startedImportingEntity();
   Decl *Result = importDeclImpl(ClangDecl, version, TypedefIsSuperfluous,
                                 HadForwardDeclaration);
-  if (!Result) {
-    ImportedDecls[{Canon, version}] = nullptr;
-    return nullptr;
-  }
 
-  if (TypedefIsSuperfluous) {
+  if (Result && TypedefIsSuperfluous) {
     SuperfluousTypedefs.insert(Canon);
     if (auto tagDecl = dyn_cast_or_null<clang::TagDecl>(Result->getClangDecl()))
       DeclsWithSuperfluousTypedefs.insert(tagDecl);
   }
 
-  if (!HadForwardDeclaration)
-    ImportedDecls[{Canon, version}] = Result;
+  if (!HadForwardDeclaration) {
+    auto it = ImportedDecls.try_emplace({Canon, version}, Result);
```

Comment:
```suggestion
    auto [it, inserted] = ImportedDecls.try_emplace({Canon, version}, Result);
```
References to variables need to be updated as well, of course. The exact names can be bikeshed, I just want to avoid multiple firsts and seconds since they are hard to keep track of.

Reply:
Done. Note that I had to separately bind `it->second` outside of the lambda due to the following warning:

```
Captured structured bindings are a C++20 extension [-Wc++20-extensions]
```

Reply:
If you move it one more line up you can use it in the if-statement also :)